src/code/TwoHeadArena.c

```diff
@@ -76,18 +76,18 @@ void* THA_GetTail(TwoHeadArena* tha) {
     return tha->tail;
 }
 
-void* THA_AllocStart(TwoHeadArena* tha, u32 size) {
+void* THA_AllocStart(TwoHeadArena* tha, size_t size) {
     void* start = tha->head;
 
-    tha->head = (u32)tha->head + size;
+    tha->head = (uintptr_t)tha->head + size;
     return start;
 }
 
 void* THA_AllocStart1(TwoHeadArena* tha) {
     return THA_AllocStart(tha, 1);
 }
 
-void* THA_AllocEnd(TwoHeadArena* tha, u32 size) {
+void* THA_AllocEnd(TwoHeadArena* tha, size_t size) {
     u32 mask;
```

Comment:
mask needs to be uintptr_t to not truncate the pointer below when it is logical AND'd.